pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -473,3 +474,12 @@ func (a *ApicastOptionsProvider) setTopologySpreadConstraints() {
 		a.apicastOptions.TopologySpreadConstraintsProduction = a.apimanager.Spec.Apicast.ProductionSpec.TopologySpreadConstraints
 	}
 }
+
+func (a *ApicastOptionsProvider) setPodTemplateAnnotations() {
+	if a.apimanager.Spec.Apicast.StagingSpec.Annotations != nil {
```

Comment:
I wonder if this condition is necessary. wdyt?

Reply:
sorry for delay, and thanks for comment. I will check

Reply:
Updated all components. Yes, this condition is not necessary.
Separate commit: "THREESCALE-9000 - annotations via CR - Update for Review -2"
But it's not all, as seems need rebase. Will update separately.

Reply:
I did rebase, so just one/latest commit is available now.

Reply:
retested, also regression test for Labels